lib/floe/workflow/context.rb

```diff
@@ -37,6 +37,35 @@ def state
         @context["State"]
       end
 
+      def state_started?
+        state.key?("EnteredTime")
+      end
+
+      def state_finished?
+        state.key?("FinishedTime")
+      end
+
+      # rename - kept name to explain
+      def step_nonblock_start
+        start_time = Time.now.utc
+
+        execution["StartTime"] ||= start_time
+        state["Guid"]            = SecureRandom.uuid
+        state["EnteredTime"]     = start_time
+
+      end
```

Comment:
I still think that starting a state belongs in the State class even if it involves things being set in Context but this works